DRILL-6964: Implement CREATE / DROP SCHEMA commands

arina-ielchiieva opened the pull request:
Note: this PR only adds support for CREATE / DROP SCHEMA commands which allow to store and delete schema. Schema usage during querying the data will be covered in other PRs.

1. Added parser methods / handles to parse CREATE / DROP SCHEMA commands.
2. Added SchemaProviders classes to separate ways of schema provision (file, table function).
3. Added schema parsing using ANTLR4 (lexer, parser, visitors).
4. Added appropriate unit tests.

Details in [DRILL-6964](https://issues.apache.org/jira/browse/DRILL-6964).

vvysotskyi commented on exec/java-exec/src/main/antlr4/org/apache/drill/exec/record/metadata/schema/parser/TableSchemaLexer.g4:
```diff
@@ -0,0 +1,83 @@
+TIME: 'TIME';
+DATE: 'DATE';
+TIMESTAMP: 'TIMESTAMP';
+INTERVAL: 'INTERVAL';
```
Not sure that just interval is enough. Maybe we should also add its qualifiers i.e. `YEAR`, `MONTH`, `DAY`, `HOUR`, `MINUTE`, `SECOND` to specify which time units should be used for underlying data.

arina-ielchiieva commented:
@vvysotskyi thanks for the review. I have address code review comments and also made a couple of new changes after Schema Provision project review on the last Hangout:
1. Removed TABLE from schema commands (CREATE TABLE SCHEMA -> CREATE SCHEMA, DROP TABLE SCHEMA -> DROP SCHEMA), renamed all appropriate classes.
2. Added TABLE to FOR clause (FOR dfs.tmp.t -> FOR TABLE dfs.tmp.t).
3. Added LOAD clause and logic for it. LOAD is used to substitute columns and their properties enumeration in the CREATE SCHEMA command.
4. Added versioning for the SchemaContainer class.

vvysotskyi commented on exec/java-exec/src/test/java/org/apache/drill/TestSchemaCommands.java:
```diff
@@ -0,0 +1,501 @@
+    String table = "dfs.tmp.table_for_simple_existing_schema";
+    try {
+      run("create table %s as select 'a' as c from (values(1))", table);
+      client.testBuilder()
```
`client.testBuilder()` -> `testBuilder()`

arina-ielchiieva commented:
@vvysotskyi thanks, addressed code review comments.
